src/middlewared/middlewared/plugins/pool_/dataset_recordsize.py

```diff
@@ -0,0 +1,31 @@
+def generate_recordsize_choices():
+    with open('/sys/module/zfs/parameters/zfs_max_recordsize') as f:
+        val = int(f.read().strip())
+        choices = []
+        for raw_val, human_readable in MAPPING.items():
+            if int(raw_val) <= val:
+                choices.append(human_readable)
+        return choices
+
+
+# https://openzfs.github.io/openzfs-docs/Performance%20and%20Tuning/Module%20Parameters.html#zfs-max-recordsize
+# Maximum supported (at time of writing) is 16MB.
+MAPPING = {
+    f'{2 << 8}': '512B',
```

Comment:
I am just curios why 2 << 8, not 1 << 9?

Reply:
> We're reading the module parameter on every update and create request or am I missing something?

Current code reads this parameter only once (on middleware startup)

Reply:
Just because of "symmetry" I guess. Left side of << are 2 which looks nice. I can change to 1<< 9 🙂

Reply:
> > We're reading the module parameter on every update and create request or am I missing something?
> 
> 
> 
> Current PR code reads this parameter only once (on middleware startup)

Then I have misunderstood how the schema variables are instantiated on each incoming request. My changes aren't correct so I'll make changes to bring them inside the method and check them there.